src/lib/s3Scan.ts

```
@@ -0,0 +1,77 @@
+    public static Copy() {
+        //Todo: Implentation Left
+    }
+
+    public static Delete() {
+        //Todo: Implentation Left
+    }
+
+    public static Purge(s3Url: string, options: {[key: string]: string | boolean | http.Agent}, callback: (err?: Error) => any): S3ScanPurgeReturn {
+        //Todo: Implentation Left
+        const readableStream = new Stream.Readable()
+        const writableStream = new Stream.Transform()
+
+        writableStream._write = (chunk, encoding, next) => {
+            console.log(chunk.toString())
+            console.log(encoding.toString())
+            next()
+        }
+
+        readableStream.pipe(writableStream)
+
+        readableStream.push(s3Url)
+        readableStream.push(JSON.stringify(options))
+        readableStream.push(null)
+
+        callback()
+        return {
+            rate: () => 0,
+            deletedCount: 0,
+            stream: writableStream
+        }
+    }
+
+    public static Get() {
+        //Todo: Implentation Left
+    }
+
+    public static List(s3Url: string, options: { [key: string]: http.Agent }): Stream.Readable {
+        //Todo: Implentation Left
+        try {
+            const readableStream = new Stream.Readable()
+            readableStream.push(`s3URL: ${s3Url}`)
+            readableStream.push(`options: ${JSON.stringify(options)}`)
+            readableStream.push(null)
+            return readableStream
+        } catch (err) {
+            throw(err)
+        }
+    }
+
+    public static Scan(s3Url: string, options: {[key: string]: string | boolean | http.Agent}): Stream.Readable {
+        //Todo: Implementation Left
+        try {
+            const readableStream = new Stream.Readable()
+            readableStream.push(`s3URL: ${s3Url}`)
+            readableStream.push(`options: ${JSON.stringify(options)}`)
+            readableStream.push(null)
+            return readableStream
+        } catch (err) {
+            throw(`error recieved: ${err}`)
+        }
```

There's a typo in the word "received".

Would it make sense to drop the try-catch block entirely and let the error bubble up naturally. It doesn't seem like the catch section contributes much here.

---

I was just templated it for now, this is going to go during the implementation